src/components/calcite-stepper-item/calcite-stepper-item.tsx

```diff
@@ -0,0 +1,255 @@
+    this.itemContent = this.getItemContent();
+    this.registerStepperItem();
+    if (this.active) this.emitRequestedItem();
+  }
+
+  componentDidUpdate() {
+    if (this.active) this.emitRequestedItem();
+  }
+
+  render() {
+    const dir = getElementDir(this.el);
+    return (
+      <Host
+        dir={dir}
+        tabindex={this.disabled ? null : 0}
+        aria-expanded={this.active.toString()}
+        onClick={() => this.emitRequestedItem()}
+      >
+        <div class="stepper-item-header">
+          {this.icon ? this.setIcon() : null}
+          {this.numbered ? (
+            <div class="stepper-item-number">{this.getItemPosition() + 1}.</div>
+          ) : null}
+          <div class="stepper-item-header-text">
+            <span class="stepper-item-title">{this.itemTitle}</span>
+            <span class="stepper-item-subtitle">{this.itemSubtitle}</span>
+          </div>
+        </div>
+        <div class="stepper-item-content">
+          <slot />
+        </div>
+      </Host>
+    );
+  }
+
+  //--------------------------------------------------------------------------
+  //
+  //  Event Listeners
+  //
+  //--------------------------------------------------------------------------
+
+  @Listen("keydown") keyDownHandler(e) {
+    if (!this.disabled && e.target === this.el) {
+      switch (e.keyCode) {
+        case SPACE:
+        case ENTER:
+          this.emitRequestedItem();
+          e.preventDefault();
+          break;
+        case UP:
+        case DOWN:
+        case LEFT:
+        case RIGHT:
+        case HOME:
+        case END:
+          this.calciteStepperItemKeyEvent.emit({ item: e });
+          e.preventDefault();
+          break;
+      }
+    }
+  }
+
+  @Listen("calciteStepperItemHasChanged", { target: "parent" })
+  updateActiveItemOnChange(event: CustomEvent) {
+    this.activePosition = event.detail.position;
+    this.determineActiveItem();
+  }
+
+  //--------------------------------------------------------------------------
+  //
+  //  Private State/Props
+  //
+  //--------------------------------------------------------------------------
+  /** position within parent */
+  private itemPosition: number;
+
+  /** the latest requested item position*/
+  private activePosition: number;
+
+  /** the slotted item content */
+  private itemContent: HTMLElement[];
+
+  //--------------------------------------------------------------------------
+  //
+  //  Private Methods
+  //
+  //--------------------------------------------------------------------------
+
+  private setIcon() {
+    var path = this.active
+      ? "circle"
+      : this.error
+      ? "exclamationMarkCircle"
+      : this.complete
+      ? "checkCircle"
+      : "circle";
+
+    // todo when calcite-icon is updated
+    // remove this and use circle-filled icon name
+    var filled = this.error || this.complete || this.active;
+
+    return (
+      <calcite-icon
+        icon={path}
+        filled={filled}
```

Comment:
this probably won't work for must icons now.

Reply:
Yeah I have a todo to update that... was Calcite UI updated to support "checkCircleF" syntax?

Reply:
no, not yet, that's on my TODO for this week to fix all that stuff in icons

Reply:
Right now most of the filled are just gone, but it uses the same syntax

Reply:
Roger. I think the three I use here are all ones that still have filled versions, I'll just update the rendered icon to use strings instead of 'filled' attribute when the time comes.

Reply:
wooord